server/connectors/confluence_connector/client.py

```
@@ -74,6 +76,119 @@ def parse_confluence_page_id(page_url: str) -> Optional[str]:
+                i += 1
+            i += 1  # skip closing ```
+            if lang:
+                raw_body = "\n".join(raw_code_lines)
+                html_parts.append(
+                    f'<ac:structured-macro ac:name="code">'
+                    f'<ac:parameter ac:name="language">{html.escape(lang)}</ac:parameter>'
+                    f'<ac:plain-text-body><![CDATA[{raw_body}]]></ac:plain-text-body>'
+                    f'</ac:structured-macro>'
+                )
+            else:
+                escaped_body = "\n".join(html.escape(l) for l in raw_code_lines)
```

Ruff (0.15.2)

[error] Ambiguous variable name: `l`

(E741)